aws-rds-dbparametergroup/src/main/java/software/amazon/rds/dbparametergroup/BaseHandlerStd.java

```diff
@@ -173,7 +174,7 @@ private ProgressEvent<ResourceModel, CallbackContext> resetParameters(final Prog
                                                                           final RequestLogger requestLogger) {
         ResourceModel model = progress.getResourceModel();
         CallbackContext callbackContext = progress.getCallbackContext();
-        Map<String, Parameter> parametersToReset = getParametersToReset(model, defaultEngineParameters, currentDBParameters);
+        Map<String, Parameter> parametersToReset = new TreeMap<>(getParametersToReset(model, defaultEngineParameters, currentDBParameters)); // TreeMap enforces sorted order on parameterName
```

Comment:
This is implementation details, please move it inside `getParametersToReset`. The invoker should not know the specifics of the underlying DS in use: it is "some map".